src_c/transform.c

```diff
@@ -570,6 +570,96 @@ scale_to(pgSurfaceObject *srcobj, pgSurfaceObject *dstobj, int width,
     return retsurf;
 }
 
+SDL_Surface *
+grayscale(pgSurfaceObject *srcobj, pgSurfaceObject *dstobj)
+{
+    SDL_Surface *src = pgSurface_AsSurface(srcobj);
+    SDL_Surface *newsurf;
+
+    if (!dstobj) {
+        newsurf = newsurf_fromsurf(src, srcobj->surf->w, srcobj->surf->h);
+        if (!newsurf)
+            return NULL;
+    }
+    else {
+        newsurf = pgSurface_AsSurface(dstobj);
+    }
+
+    if (newsurf->w != src->w || newsurf->h != src->h) {
+        return (SDL_Surface *)(RAISE(
+            PyExc_ValueError,
+            "Destination surface must be the same size as source surface."));
+    }
+
+    if (src->format->BytesPerPixel != newsurf->format->BytesPerPixel) {
+        return (SDL_Surface *)(RAISE(
+            PyExc_ValueError,
+            "Source and destination surfaces need the same format."));
+    }
+
+    Uint32 *pixels = (Uint32 *)src->pixels;
```

Comment:
This assumes the bit depth is 32 pixels.

If you give it a 24 bit surface or any other depth, it messes up.

32 bit:
![output32](https://user-images.githubusercontent.com/46412508/201453074-6e708988-1876-4c53-ba3d-4b48504f58d9.png)

24 bit:
![output24](https://user-images.githubusercontent.com/46412508/201453089-cf047b59-e759-4d61-ac1f-528c54ac87a6.png)